pkg/controller/constraint/constraint_controller.go

```diff
@@ -338,20 +335,44 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 					}
 					currentVapBinding = nil
 				}
-				newVapBinding := &admissionregistrationv1beta1.ValidatingAdmissionPolicyBinding{}
+				newVapBinding := &unstructured.Unstructured{}
+				newVapBinding.SetGroupVersionKind(schema.GroupVersionKind{Group: groupVersion.Group, Version: groupVersion.Version, Kind: "ValidatingAdmissionPolicyBinding"})
 				transformedVapBinding, err := transform.ConstraintToBinding(instance, VAPEnforcementActions)
 				if err != nil {
 					status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
 					if err2 := r.writer.Update(ctx, status); err2 != nil {
-						log.Error(err2, "could not report transform vapbinding error status")
+						log.Error(err2, "could not report transform vapbinding error", "vapName", vapBindingName)
 					}
 					return reconcile.Result{}, err
 				}
 				if currentVapBinding == nil {
-					newVapBinding = transformedVapBinding.DeepCopy()
+					err = r.scheme.Convert(transformedVapBinding, newVapBinding, nil)
+					if err != nil {
+						status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+						if err2 := r.writer.Update(ctx, status); err2 != nil {
+							log.Error(err2, "could not report convert vapbinding error", "vapName", vapBindingName)
+						}
+						return reconcile.Result{}, err
+					}
+					newVapBinding.SetGroupVersionKind(schema.GroupVersionKind{Group: groupVersion.Group, Version: groupVersion.Version, Kind: "ValidatingAdmissionPolicyBinding"})
 				} else {
 					newVapBinding = currentVapBinding.DeepCopy()
-					newVapBinding.Spec = transformedVapBinding.Spec
+					unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&transformedVapBinding.Spec)
```

Comment:
Lets just separate this out to a separate function and use typecasting rather than resorting to unstructured.